File: compileanalysis.m
function compileCSVFiles()
    % COMPILECSVFILES Compile multiple CSV files into a single file for each replicate
    % This function allows the user to select multiple text files. Each text file
    % contains a list of paths to CSV files. The function reads the CSV files,
    % combines their content, and saves the combined data into a new CSV file
    % for each input text file.

    % Use uigetfile to open a file dialog for selecting multiple text files
    [fileNames, filePath] = uigetfile('*.csv', 'Select compiled Fixtrax paths file for multiple replicates (Multiple can be selected)', 'MultiSelect', 'on');
    
    % Check if multiple files are selected
    if iscell(fileNames)
        inputFiles = fullfile(filePath, fileNames);
    else
        inputFiles = {fullfile(filePath, fileNames)};
    end

    % Loop through each selected input file
    for i = 1:length(inputFiles)
        inputFile = inputFiles{i};
        disp(inputFile);
        
        % Read the list of CSV file paths from the text file into a MATLAB cell array
        fid = fopen(inputFile, 'r');
        csvFilePaths = textscan(fid, '%s', 'Delimiter', '\n');
        fclose(fid);
        
        csvFilePaths = csvFilePaths{1};
        
        % Create an empty cell array to store the tables
        dataTables = {};

        % Iterate over the list of CSV file paths and read each CSV file into a table
        for j = 1:length(csvFilePaths)
            csvFilePath = strtrim(csvFilePaths{j}); % Remove any leading/trailing whitespace
            csvFile = fullfile(csvFilePath, 'Analyzed_scores.csv');
            dataTable = readtable(csvFile);
            dataTables{end+1} = dataTable; %#ok<*AGROW>
        end

        % Concatenate the tables using the MATLAB vertcat function
        combinedTable = vertcat(dataTables{:});

        % Generate the output file name
        [~, name, ~] = fileparts(inputFile);
        outputFile = fullfile(filePath, [name '_Compiled.csv']);

        % Save the combined table to a new CSV file
        writetable(combinedTable, outputFile);
    end
end

File: Old python codes/compileanalysis.py
import pandas as pd
from tkinter import *                     
from tkinter import filedialog as fd     
import os    

def openFile():
    the_file = fd.askopenfilenames(title = "Select compiled Fixtrax paths file for multiple replicates (Multiple can be selected)")
    return the_file


input_files= openFile()

for input_file in input_files:

    print(input_file)

    # Read the list of CSV file paths from the text file into a Python list.
    with open(input_file, "r") as f:
        csv_file_paths = f.readlines()

    # Create an empty list to store the DataFrames.
    dataframes = []

    # Iterate over the list of CSV file paths and read each CSV file into a DataFrame.
    for csv_file_path in csv_file_paths:
        dataframe = pd.read_csv(csv_file_path.rstrip('\n')+'\Analyzed_scores.csv')
        dataframes.append(dataframe)

    # Concatenate the DataFrames using the Pandas `concat()` function.
    combined_dataframe = pd.concat(dataframes)

    output_file = input_file.removesuffix(".csv") + "_Compiled.csv"

    # Save the combined DataFrame to a new CSV file.
    combined_dataframe.to_csv(output_file, index=False)
